common/src/riscv/encoder.rs

```diff
@@ -118,16 +124,23 @@ pub fn encode_instruction(instruction: &Instruction) -> u32 {
             InstructionType::UType => encode_u_type(instruction).to_le(),
             InstructionType::JType => encode_j_type(instruction).to_le(),
             InstructionType::Unimpl => 0,
-        }
+        };
+        Ok(encoded)
     } else {
-        // TODO: handle built-in custom instructions.
-        //
-        // The only supported opcode is keccakf.
-        assert_eq!(instruction.opcode.raw, KECCAKF_OPCODE);
-        encode_s_type(instruction).to_le()
+        if instruction.opcode.raw != KECCAKF_OPCODE {
+            return Err(EncodeError::UnsupportedCustomOpcode(
+                instruction.opcode.raw.into(),
+            ));
+        }
+        Ok(encode_s_type(instruction).to_le())
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Eq)]
```

Comment:
Why not use the existing `OpcodeError` type (https://github.com/nexus-xyz/nexus-zkvm/blob/main/common/src/error/opcode.rs#L4-L9)?